doc/stages/readers.hdf.rst

```diff
@@ -0,0 +1,101 @@
+.. _readers.hdf:
+
+readers.hdf
+===============
+
+The **HDF reader** reads from files in the HDF5 format. See the
+TODO for more information.
+
+.. plugin::
+
+.. streamable::
+
+Example
+-------
+This example reads from the Autzen HDF example with all dimension
+properly mapped and then outputs a LAS file.
+
+.. code-block:: json
+
+    [
+        {
+            "type": "readers.hdf",
+            "filename": "test/data/hdf/autzen.h5",
+            "map":
+            {
+                "X" : "autzen/X",
+                "Y" : "autzen/Y",
+                "Z" : "autzen/Z",
+                "Red" : "autzen/Red",
+                "Blue" : "autzen/Blue",
+                "Green" : "autzen/Green",
+                "Classification" : "autzen/Classification",
+                "EdgeOfFlightLine" : "autzen/EdgeOfFlightLine",
+                "GpsTime" : "autzen/GpsTime",
+                "Intensity" : "autzen/Intensity",
+                "NumberOfReturns" : "autzen/NumberOfReturns",
+                "PointSourceId" : "autzen/PointSourceId",
+                "ReturnNumber" : "autzen/ReturnNumber",
+                "ScanAngleRank" : "autzen/ScanAngleRank",
+                "ScanDirectionFlag" : "autzen/ScanDirectionFlag",
+                "UserData" : "autzen/UserData"
+            }
+        },
+        {
+            "type" : "writers.las",
+            "filename": "output.las"
+        }
+    ]
+
+
+.. note::
+    All dimensions must be simple numeric HDF datasets with
+    equal lengths. Compound types, enum types, string types,
+    etc. are not supported.
+
+
+.. warning::
+    The HDF reader does not set an SRS.
+
+
+Common Use Cases
+----------------
+
+A common use case for this data is reading NASA's ICESAT2 data.
+This example reads the X, Y, and Z coordinates from the ICESAT2
+ATL03 format and converts them into a LAS file.
+
+.. note::
+    ICESAT2 data use EPSG:7912.
+
+.. code-block:: json
+
+    [
+        { 
+            "type": "readers.hdf", 
+            "filename": "ATL03_20190906201911_10800413_002_01.h5",  
+            "map": 
+            { 
+                "X" : "gt1l/heights/lon_ph", 
+                "Y" : "gt1l/heights/lat_ph", 
+                "Z" : "gt1l/heights/h_ph", 
+            } 
+        }, 
+        { 
+            "type" : "writers.las", 
+            "filename": "output.las" 
```

Comment:
You should set ``scale_x``, ``scale_y``, and ``scale_z`` here along with auto on the offsets. A user following this example is going to get trimmed data due to the XYs being Lat/lon.